cmd/launcher/launcher.go

```diff
@@ -168,13 +170,49 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 	flagController := flags.NewFlagController(logger, stores[storage.AgentFlagsStore], fcOpts...)
 	k := knapsack.New(stores, flagController, db)
 
+	slogLevel := slog.LevelInfo
+	if k.Debug() {
+		slogLevel = slog.LevelDebug
+	}
+
+	slogAttrReplacementFunc := func(groups []string, a slog.Attr) slog.Attr {
+		if a.Key != slog.TimeKey {
+			return a
+		}
+
+		return slog.Attr{
+			Key:   slog.TimeKey,
+			Value: slog.AnyValue(time.Now().UTC()),
+		}
+	}
+
+	k.AddLogHandler(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
+		AddSource:   true,
+		Level:       slogLevel,
+		ReplaceAttr: slogAttrReplacementFunc,
+	}))
```

Comment:
I'm not sure if we want that, and TBH, it reminds me of a thing I never really solved in #1322. To wit:

In today's world, we log info the os facilities, and debug to internal logs. _And_ when running from the command line, during devel, we usually want debug to stderr. _But_ because on macOS and linux stderr is the system log, this leaves us with a bit of a gordian knot.

I think my idea world has something like:
- all logs written to internal debug logs
- all logs sent to logshipper, though maybe only shipped if the control server says so
- Some startup errors sent to system logs

But that leaves an question about how to handle dev usage.

I wonder if we should slightly repurpose the `debug` flag. When it's present, we copy all logs to the stderr. And when it's not, only the startup logs. Eg, kinda making it something like "I am a developer running this on the command line"

Reply:
I often run launcher in the foreground when debugging k2. (And I bet k2 folks do as well) And in that use case, I very much want my logs in my terminal.

Reply:
not sure if leveling is the right approach, but we could create a systemLevel and add a handler that write to stderr only when someone uses system level

Reply:
I went ahead and added this, just to see how it felt, was simple